src/simplify_algebra.cpp

```diff
@@ -714,6 +716,132 @@ struct find_rsqrt
     }
 };
 
+template <class Op>
+static bool check_dims(const std::string& name,
+                       std::vector<instruction_ref>& vec_ins,
+                       const std::vector<int64_t>& dims)
+{
+    return std::all_of(vec_ins.begin(), vec_ins.end(), [&](auto i) {
+        return (i->name() == name) and (any_cast<Op>(i->get_operator()).dims == dims);
```

Comment:
There is no need to pass in the dims. This can be written as:

```cpp
static bool same_ops(const std::vector<instruction_ref>& vec_ins) {
    return std::all_of(vec_ins.begin(), vec_ins.end(), [&](auto i) {
        return i->get_operator() == vec_ins.front()->get_operator();
    };
}
```

The default equality operator will compare both the name and the dims.

Reply:
fixed